thirdparties/glog/glog.patch

```diff
@@ -741,7 +742,30 @@ index 0000000..1f3c8c1
 +    }
 +}
 +
++// NOTE: this function only run once in child process
++void AsyncLogger::ResetInChildProcess(bool start) {
++    if (!PidHasChanged()) {
++        return;
++    }
++
++    // After fork(2) and in the child process we should:
++    //   (1) reset all status for AyncLogger
++    //   (2) restart AyncLogger when necessary
++    appThreadsBlockedCount_ = 0;
++    flushCount_ = 0;
++    mutex_.unlock();
```

Comment:
according to the description in here https://en.cppreference.com/w/cpp/thread/mutex/unlock, 

```
The mutex must be locked by the current thread of execution, otherwise, the behavior is undefined.
```

and, destroy a mutex locked by another thread is also UB.
https://en.cppreference.com/w/cpp/thread/mutex/%7Emutex

so, current solution may not work well.

you can check whether `pthread_mutexattr_setpshared` is helpful, or find another way.

Reply:
> 

yep, it's a bad solution, i have changed a way to solve it.